packages/itwinui-react/src/core/Tabs/Tabs.tsx

```diff
@@ -64,547 +43,759 @@ type TabsOrientationProps =
-      return React.isValidElement(tab) && tab.props.disabled;
-    };
+    const nextTab = (tabRef.current?.nextElementSibling ??
+      allTabs.at(0)) as HTMLElement;
 
-    let newIndex = focusedIndex ?? currentActiveIndex;
-
-    /** focus next tab if delta is +1, previous tab if -1 */
-    const focusTab = (delta = +1) => {
-      do {
-        newIndex = (newIndex + delta + labels.length) % labels.length;
-      } while (isTabDisabled(newIndex) && newIndex !== focusedIndex);
-      setFocusedIndex(newIndex);
-      focusActivationMode === 'auto' && onTabClick(newIndex);
-    };
+    const previousTab = (tabRef.current?.previousElementSibling ??
+      allTabs.at(-1)) as HTMLElement;
 
     switch (event.key) {
       case 'ArrowDown': {
         if (orientation === 'vertical') {
-          focusTab(+1);
+          nextTab?.focus();
           event.preventDefault();
         }
         break;
       }
       case 'ArrowRight': {
         if (orientation === 'horizontal') {
-          focusTab(+1);
+          nextTab?.focus();
           event.preventDefault();
         }
         break;
       }
       case 'ArrowUp': {
         if (orientation === 'vertical') {
-          focusTab(-1);
+          previousTab?.focus();
           event.preventDefault();
         }
         break;
       }
       case 'ArrowLeft': {
         if (orientation === 'horizontal') {
-          focusTab(-1);
+          previousTab?.focus();
           event.preventDefault();
         }
         break;
       }
-      case 'Enter':
-      case ' ':
-      case 'Spacebar': {
-        event.preventDefault();
-        if (focusActivationMode === 'manual' && focusedIndex !== undefined) {
-          onTabClick(focusedIndex);
-        }
-        break;
-      }
       default:
         break;
     }
   };
 
-  const createTab = React.useCallback(
-    (label: React.ReactNode, index: number) => {
-      const onClick = () => {
-        setFocusedIndex(index);
-        onTabClick(index);
-      };
-      return (
-        <li key={index}>
-          {!React.isValidElement(label) ? (
-            <Tab
-              label={label}
-              className={cx({
-                'iui-active': index === currentActiveIndex,
-              })}
-              tabIndex={index === currentActiveIndex ? 0 : -1}
-              onClick={onClick}
-              aria-selected={index === currentActiveIndex}
-            />
-          ) : (
-            React.cloneElement(label as JSX.Element, {
-              active: index === currentActiveIndex,
-              'aria-selected': index === currentActiveIndex,
-              tabIndex: index === currentActiveIndex ? 0 : -1,
-              onClick: (args: unknown) => {
-                onClick();
-                label.props.onClick?.(args);
-              },
-            })
-          )}
-        </li>
-      );
+  // use first tab as active if no `isActive` passed.
+  const setInitialActiveRef = React.useCallback(
```

Comment:
Can this not be called in a `useEffect`? Is there an advantage with passing it in a `useCallback()` and passing that to `useMergedRefs`?

Reply:
We need the first tab. This way is more stable to keep track of it.